pennylane/templates/subroutines/select_pauli_rot.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the SelectPauliRot template.
+"""
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+from pennylane.templates.state_preparations.mottonen import _apply_uniform_rotation_dagger
+
+
+class SelectPauliRot(Operation):
+    r"""Applies the multiplexer with Pauli rotations.
+
+    This operator, also called a "multiplexed rotation", applies a sequence of uniformly controlled
+    rotations to a target qubit. The rotations are selected based on the values encoded in the control qubits.
+    Its definition is given by:
+
+    .. math::
+
+       \sum_i | i \rangle \langle i | \otimes R(\alpha_i)
+
+    Here, :math:`| i \rangle` refers to the computational basis state of the control register,
+    and :math:`R(\cdot)` denotes a Pauli rotation applied to the target qubit,
+    parametrized by :math:`\alpha_i`.
+
+    For more details, see `Möttönen and Vartiainen (2005), Fig 7a <https://arxiv.org/abs/quant-ph/0504100>`_.
+
+    **Example**
+
+    .. code-block::
+
+        angles = np.array([1.0, 2.0, 3.0, 4.0])
+
+        wires = qml.registers({"control": 2, "target": 1})
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.SelectPauliRot(
+                angles,
+                control_wires=wires["control"],
+                target_wire=wires["target"],
+                rot_axis="Y",
+            )
+            return qml.state()
+
+    .. code-block:: pycon
+
+        >>> print(circuit())
+        [0.87758256+0.j 0.47942554+0.j 0.        +0.j 0.        +0.j
+         0.        +0.j 0.        +0.j 0.        +0.j 0.        +0.j]
+
+    .. seealso:: :class:`~.Select`.
+
+    Args:
+        angles (tensor_like): The rotation angles to be applied.
+        control_wires (Sequence[int]): The control qubits used to select the rotation.
+        target_wire (Sequence[int]): The wire where the rotations are applied.
+        rot_axis (str): The axis around the rotation is performed.
+            It can take the value ``X``, ``Y`` or ``Z``. Default is ``Z``.
+
+    Raises:
+        ValueError: If the length of the angles array is not :math:`2^n`, where :math:`n` is the number
```

Comment:
Might be good to add this to the Args as well, e.g., 
angles (tensor_like): The rotation angles to be applied. The length of the angles array must be :math:`2^n`, where :math:`n` is the number of ``control_wires``.